src/sentry/api/fields/secret.py

```diff
@@ -0,0 +1,52 @@
+from typing import Dict, Optional, Union
+
+from rest_framework import serializers
+
+from sentry.utils import json
+
+
+class SecretField(serializers.Field):  # type: ignore
+    """
+    A secret-containing field whose values are either a string or a magic object. This validator
+    should be used when secrets are known to have previous values already stored on the server, i.e.
+    they may be in the form of a magic object representing a redacted secret.
```

Comment:
Very much irrelevant nitpicking: but some PEP (257?) says you should start docstrings with a single line followed by a blank line.  And should be written imperatively (already done).